TTS/auto_tts/complete_recipes.py

```diff
@@ -0,0 +1,309 @@
+import zipfile
+
+import requests
+import tqdm
+
+from TTS.auto_tts.model_hub import TtsModels, VocoderModels
+from TTS.auto_tts.utils import data_loader
+from TTS.trainer import Trainer, TrainingArgs, init_training
+
+
+class TtsAutoTrainer(TtsModels):
+    """
+    Args:
+
+        data_path (str):
+            The path to the dataset. Defaults to None.
+
+        dataset (str):
+            The dataset identifier. ex: ljspeech would be "ljspeech". Defaults to None.
+            See auto_tts utils for specific dataset names.
+
+        batch_size (int):
+            The size the batches you pass to the model. This will depend on gpu memory.
+            less than 32 is not recommended. Defaults to 32.
+
+        output_path (str):
+            The path where you want to the model config and model weights. If it is None it will
+            use your current directory. Defaults to None
+
+        mixed_precision (bool):
+            enables mixed precision training. can make batch sizes bigger and make training faster.
+            Could also make some trainings unstable. Defualts to False.
+
+        learning_rate (float):
+            The learning rate for the model. Defaults to 1e-3.
+
+        epochs (int):
+            how many times you want to model to go through the entire dataset. This usually doesn't need changing.
+            Defaults to 1000.
+
+    Usage:
+        Python:
+            From TTS.auto_tts.complete_recipes import TtsAutoTrainer
+            trainer = TtsAutoTrainer(data_path='DEFINE THIS', stats_path=None, dataset="DEFINE THIS" batch_size=32, learning_rate=0.001,
+                      mixed_precision=False, output_path='DEFINE THIS', epochs=1000)
+            model = trainer.single_speaker_autotts("tacotron2, "double decoder consistency")
+            model.fit()
+
+        command line:
+            python single_speaker_autotts.py --data_path ../LJSpeech-1.1 --dataset ljspeech --batch_size 32 --mixed_precision
+            --model tacotron2 --tacotron2_model_type double decoder consistency --forward_attention
+            --location_attention
+
+    """
+
+    def __init__(
+        self,
+        data_path=None,
+        dataset=None,
+        batch_size=32,
+        output_path=None,
+        mixed_precision=False,
+        learning_rate=1e-3,
+        epochs=1000,
+    ):
+
+        super().__init__(batch_size, mixed_precision, learning_rate, epochs, output_path)
+        self.data_path = data_path
+        self.dataset_name = dataset
+
+    def single_speaker_autotts(  # im actually going to change this to autotts_recipes and i'm making a more generic
```

Comment:
For docstrings use this format https://numpydoc.readthedocs.io/en/latest/format.html

Reply:
Even for personal notes